### PR TITLE
fix thread unsafe problem in server-alarm-plugin (#4230)

### PR DESCRIPTION
**Pull request by liuhaoXD**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [x] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues
#4230
___
### Bug fix
- Bug description.
fix thread-unsafe operation

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

**Comment by wu-sheng** [comment]
Two suggestions.
1. `#add` should not include `#moveTo`, as in `#in`, we move the time window first, then call add. 
1. This method should only be called by a single thread. So basically, no concurrency access. This lock is for fail-safe.

Could you follow these to update this PR?

**Comment by wu-sheng** [comment]
Any update?

**Comment by wu-sheng** [comment]
CI fails, please recheck.

File: oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RunningRule.java [hunk 1]
```diff
@@ -187,7 +182,7 @@ public void moveTo(LocalDateTime targetTime) {
         private int silenceCountdown;
 
         private LinkedList<Metrics> values;
-        private ReentrantLock lock = new ReentrantLock();
+        private ReentrantLock reentrantLock = new ReentrantLock();
```
Comment (wu-sheng) [inline oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RunningRule.java]: Why change this name?

File: oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RunningRule.java [hunk 2]
```diff
@@ -86,7 +83,7 @@ public RunningRule(AlarmRule alarmRule) {
      * Receive metrics result from persistence, after it is saved into storage. In alarm, only minute dimensionality
      * metrics are expected to process.
      *
-     * @param meta of input metrics
+     * @param meta    of input metrics
```
Comment (wu-sheng) [inline oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RunningRule.java]: Why change this?

File: oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RunningRule.java [hunk 3]
```diff
@@ -290,28 +278,28 @@ private boolean isMatch() {
...
-                        double dvalue = ((DoubleValueHolder)metrics).getValue();
+                        double dvalue = ((DoubleValueHolder) metrics).getValue();
                         double dexpected = RunningRule.this.threshold.getDoubleThreshold();
                         if (op.test(dexpected, dvalue)) {
                             matchCount++;
                         }
                         break;
                     case MULTI_INTS:
-                        int[] ivalueArray = ((MultiIntValuesHolder)metrics).getValues();
+                        int[] ivalueArray = ((MultiIntValuesHolder) metrics).getValues();
```
Comment (wu-sheng) [inline oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RunningRule.java]: What change the format of all these cases?

**Comment by liuhaoXD** [comment]
> Any update?

Actually, I think that `#moveTo` when doing `#in` is redundant, which can be covered by `#add` operation, and `#moveTo` in `#add` is an essential mechanism to make `public void add()` more safe at `window`'s initial stage, another PR will be created today :).

File: oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RunningRule.java (hunk 3 above)
Comment (liuhaoXD) [reply oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RunningRule.java]: alibaba coding guideline plugin did these coding format automatically..

Comment (wu-sheng) [reply oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RunningRule.java]: But we are not using it .SkyWalking has its own format, please revert those.

**Comment by wu-sheng** [comment]
> Actually, I think that #moveTo when doing #in is redundant

I think you have removed it.

**Comment by wu-sheng** [comment]
The PR could be accepted when you make all CI passed. Please fix them.

**Comment by wu-sheng** [comment]
@liuhaoXD Could you follow another two format change requests?

**Comment by wu-sheng** [comment]
If this merge in time, you will be our 200th contributor on this main repo.

File: oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RunningRule.java [hunk 4]
```diff
@@ -227,18 +221,17 @@ public void moveTo(LocalDateTime current) {
...
-
-            }
+            LocalDateTime timeBucket = TIME_BUCKET_FORMATTER.parseLocalDateTime(bucket + "");
 
-            lock.lock();
+            this.lock.lock();
             try {
+                if (this.endTime == null) {
+                    init();
+                    this.endTime = timeBucket;
```
Comment (wu-sheng) [inline oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RunningRule.java]: Why add this? End time has been initialized in the constructor. End time should never be null. Right?

**Comment by kezhenxu94** [comment]
> LGTM. Format issues related to #4272. I will ignore this for now.
> 
> @kezhenxu94 Recheck?

Will check later

**Pull request merged**